fibonacci.asm

```diff
@@ -1,88 +1,86 @@
 ##########################################################
+.data
+fibs: .word 0 : 9 # create an array variable named "fibs" of 9 word-length elements (4 bytes each)
+size: .word 9 # create a single integer variable named "size" that indicates the length of the array
 
-	.data
-
-fibs:	.word   0 : 9         # create an array variable named "fibs" of 9 word-length elements (4 bytes each)
-size:	.word  9              # create a single integer variable named "size" that indicates the length of the array
-
+# create a string to ask for user input
+string: .asciiz "How many Fibonacci numbers do you want to print? ( Enter an integer between 1 and 9 )"
 ###############################################
 # .text segment. Assembly instructions go here.
 ###############################################
-	.text
-	      la   $s0, fibs        # load address of array into $s0
-	      la   $s5, size        # load address of size variable into $s5
-	      lw   $s5, 0($s5)      # load array size from its address in the register
-
-	      li   $s2, 1           # Initialize the Fibonacci numbers with value 1, stored in $s2
-	      sw   $s2, 0($s0)      # Set fibs[0] to 1
-	      sw   $s2, 4($s0)      # Set fibs[1] to 1
-	      addi $s1, $s5, -2     # Counter for loop, will execute (size-2) times
-
-	      # Loop to compute each Fibonacci number using the previous two Fib. numbers.
-	      # On line 37, "loop" is a label
-loop:	 lw   $s3, 0($s0)      # Get value from array fibs[i-2]
-	      lw   $s4, 4($s0)      # Get value from array fibs[i-1]
-	      add  $s2, $s3, $s4    # fibs[i] = fibs[i-1] + fibs[i-2]
-	      sw   $s2, 8($s0)      # Store newly computed fibs[i] in array
-	      addi $s0, $s0, 4      # increment address for next fibonacci entry
-	      addi $s1, $s1, -1     # decrement loop counter
-	      bgtz $s1, loop        # repeat while not finished
-
-	      # Now the Fibonacci numbers are computed and stored in array. Print them.
-	      la   $a0, fibs        # first argument for print (array)
-	      add  $a1, $zero, $s5  # second argument for print (size)
-	      jal  print            # call print routine (note the 'print' label on line 68)
-
-	      # The program is finished. Exit.
-	      li   $v0, 10          # system call for exit
-	      syscall               
-
-	###############################################################
-	# Subroutine to print the numbers on one line. Another .data segment.
-	###########################################################
-	      .data
-	      
-space:	.asciiz  " "          				# Print a space between each pair of numbers
-head:	 .asciiz  "The Fibonacci numbers are:\n" 	# Print a little helpful intro
-
+.text
+
+############# BEGIN MY EDITS ##################
+
+li $v0, 4 # system call for printing string
+la $a0, string # load address of string to print in $a0
+syscall
+
+li $v0, 5 # system call to read an integer input by user
+syscall
+
+sw $v0, size # store the integer from user input into size, which allows
```

Comment:
Line 33: In other instances, may not be a good idea to write over size since it is tied to the max capacity of the array.